source/particle/property/integrated_strain.cc

```diff
@@ -31,8 +31,9 @@ namespace aspect
       IntegratedStrain<dim>::initialize_one_particle_property(const Point<dim> &,
                                                               std::vector<double> &data) const
       {
-        for (unsigned int i = 0; i < SymmetricTensor<2,dim>::n_independent_components ; ++i)
-          data.push_back(0.0);
+        const Tensor<2,dim> identity = Tensor<2,dim>(unit_symmetric_tensor<dim>());
```

Comment:
I don't think you need the extra cast on the rhs, or do you? In either case, you can probably make the variable `static`.